src/OrbitBase/include/OrbitBase/GetProcAddress.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_GET_PROC_ADDRESS_H_
+#define ORBIT_BASE_GET_PROC_ADDRESS_H_
+
+#include <Windows.h>
+
+#include <string>
+
+#include "GetLastError.h"
+#include "Logging.h"
+#include "UniqueResource.h"
+
+// clang-format off
+#include <libloaderapi.h>
+// clang-format on
+
+namespace orbit_base {
+
+inline void* GetProcAddress(const std::string& module, const std::string& function) {
+  HMODULE module_handle = LoadLibraryA(module.c_str());
+  if (module_handle == nullptr) {
+    ORBIT_ERROR("Could not load module \"%s\" while looking for function \"%s\": %s", module,
+                function, GetLastErrorAsString());
+    return nullptr;
+  }
+  orbit_base::unique_resource handle_closer(module_handle, ::FreeLibrary);
```

Comment:
I'm not sure if that is what you want. It will call `FreeLibrary` as soon as `handle_closer` gets out of scope which is at the end of this function. If that leads to an unload of the library (reference count reaches zero) the returned address will be useless.

I believe your tests don't fail because the libraries there are loaded anyway which means `FreeLibrary` only decrements the reference count here.

If you want to make this scoped you have to return a `unique_resource` and the caller of this function has to keep this object around until its not needed anymore.

Reply:
I was assuming that the library was already loaded. I removed the LoadLibrary/FreeLibrary pair in favor of GetModuleHandle which will return a nullptr handle if the library is not loaded, in which case we return an error message.

Reply:
Great! That's a nice solution!